core/src/Machine.js

```diff
@@ -22,12 +30,43 @@ export default class Machine {
     return require("bluebird").Promise;
   }
 
+  static defaultHistory(promise) {
+    return {
+      add() {
+        /* istanbul ignore next */
+        return promise.resolve({});
+      },
+      search() {
+        /* istanbul ignore next */
+        return promise.resolve([]);
+      }
+    }
+  }
+
   // sets object initial state
-  start({ object }) {
-    const { objectStateFieldName, initialState } = this.machineDefinition.schema;
+  // @param object - stateful object
+  // @param user - user name who initiated event/transition (this info will be writted into object wortkflow history)
+  // @param description - event/transition/object description (this info will be writted into object wortkflow history)
+  // N!B!: history record fileds 'from' and 'event' will be filled with 'NULL' value
+  start({ object, user, description }) {
+    const { name, objectStateFieldName, initialState } = this.machineDefinition.schema;
     // eslint-disable-next-line no-param-reassign
     object[objectStateFieldName] = initialState;
-    return this.promise.resolve({ object });
+    // add history record
+    return this.history.add({
+      from: 'NULL',
+      to: initialState,
+      event: 'NULL',
```

Comment:
Event NULL is not really good in case of starting workflow. Proposals:
* from_NULL_to_<initialState>
* START_WORKFLOW_<workflowName>

Reply:
After discussion: __START__